clang/test/CodeGen/fast-math.c

```diff
@@ -1,4 +1,4 @@
-// RUN: %clang_cc1 -ffast-math -ffp-contract=fast -emit-llvm -o - %s | FileCheck %s
+// RUN: %clang_cc1 -ffast-math -ffp-contract=fast -emit-llvm -O2 -o - %s | FileCheck %s
```

Comment:
It seems to me that this issue exists only at O2, right? https://godbolt.org/z/xGcWG1aM6
Shouldn't that be reflected in the code somewhere?

Reply:
Yes, this fact is reflected in #84648. Actually the problem is incorrect optimization caused by wrong fast-math flags.

Also your example cannot demonstrate the issue because it is C++ code. In C++ mode instructions like SelectInst or PHINode do not get fast-math flags.